source/client/RunManagement.py

```diff
@@ -0,0 +1,250 @@
+                    this_wild.tempnumber = this_wild.number
+                    groups_wilds.append(this_wild)
+                    card_group.append(card)
+                else:
+                    raise Exception('Card value already in the run.')
+            elif card.tempnumber > (abs_temp_number + 2):
+                raise Exception('too big a gap between numbers')
+            else:
+                # gap must be 1 card. It's possible to free up a wild later in run then gap of 1 can be closed.
+                gap_flag =True
+    if gap_flag and len(groups_wilds) > 0:
+        temp_run_group = card_group
+        card_group = []
+        card_group.append(temp_run_group[0])
+        for indx in range(len(temp_run_group)-1):
+            card = temp_run_group[indx+1]
+            if temp_run_group[indx].tempnumber == card.tempnumber + 1:
+                card_group.append(card)
+            elif temp_run_group[indx].tempnumber == card.tempnumber + 2:
+                if len(groups_wilds) > 0:
+                    this_wild.pop(groups_wilds[0])
+                    card_group.append(this_wild)
+                    card_group.append(card)
+                else:
+                    raise Exception('too big a gap between numbers')
+
+    #  Review note - Handle Aces after other cards, else ran into problem when wanted Ace, Joker, 3...
+    # Rare for Ace Hi and Ace low to both be options. If they are, does it make a difference which one they are?
+    # If run is A?,2,Wild,4...J,Q,K,A? then it does (wild could still be played above King, but not below 2).
+    if len(aces_list) > 0:
+        if len(aces_list) > 2:
+            raise Exception('Cannot play more than 2 Aces in a single run.')
+        # Possible Ace(s) replacing a wild card with tempnumber 13 or -1.
+        # If have wild in Ace slot then remove it from run.  It will be placed back in run after Aces placed.
+        if isWild(card_group[0], wild_numbers) and abs(card_group[0].tempnumber) == 1:
+            this_wild = card_group.pop(0)
+            groups_wilds.append(this_wild)
+        if isWild(card_group[-1], wild_numbers) and card_group[-1].tempnumber == 13:
+            this_wild = card_group.pop(-1)
+            groups_wilds.append(this_wild)
+        # Can Ace go high but not low?
+        if card_group[-1].tempnumber == 13 and not card_group[0].tempnumber == 2:
+            this_ace = aces_list.pop(0)
+            this_ace.tempnumber = 14
+            card_group.append(this_ace)
+            # Can Ace go low but not high?
+        elif not card_group[-1].tempnumber == 13 and card_group[0].tempnumber == 2:
+            this_ace = aces_list.pop(0)
+            this_ace.tempnumber = -1
+            card_group.insert(0, this_ace)
+        elif card_group[-1].tempnumber == 13 and card_group[0].tempnumber == 2:
+            if len(aces_list) == 1:
+                print("IF ACE CAN BE HIGH OR LOW THAN AUTOMATICALLY MAKING IT LOW, RATHER THAN HANDLING CORNER CASE.")
```

Comment:
Instead of checking the length you can place the low ace immediately, then check if there is another ace left with a single if (aces_list.isEmpty or equivalent) and add the high ace then.